_test/test/build_integration_test.dart

```diff
@@ -92,5 +93,23 @@ void main() {
       await runBuild(trailingArgs: ['--output', 'build']);
       expect(dartSource.existsSync(), true);
     });
+
+    test('Re-snapshots if there is no asset graph', () async {
+      var assetGraph = assetGraphPathFor(p.url
+          .join('.dart_tool', 'build', 'entrypoint', 'build.dart.snapshot'));
+      await File(assetGraph).delete();
+
+      var nextBuild = await runBuild();
+      printOnFailure(nextBuild.stdout.toString());
```

Comment:
Turns out we always do this anyway. This can be removed.
https://github.com/dart-lang/build/blob/708c302227c8fb254550883aaae25236cb8c29cc/_test/test/common/utils.dart#L63-L64